### PR TITLE
pay-respects: add module

### PR DESCRIPTION
**Pull request by ALameLlama**
### Description
Add pay-respects module including shell integration support for bash, zsh, fish and nushell

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by ALameLlama** [comment]
Closes #6204

**Comment by ALameLlama** [comment]
I'm still fairly new to nix so if anything needs changing let me know :)

I looked over how a few other modules work and put this together. the tests passed using
```
nix-shell --pure tests -A run.pay-respects-integration-enabled
nix-shell --pure tests -A run.pay-respects-integration-disabled
```

I manually tested it in ZSH in my own configuration. and used https://github.com/iffse/pay-respects to get the config for fish and nu.

**Comment by ALameLlama** [comment]
I'm not sure if @SigmaSquadron has any interest in maintaining this also as they are the ones that added into nixpkgs

**Comment by minhtrancccp** [comment]
i specified `pay-respects.package = inputs.pay-respect.packages.[system].default` (i.e. [the repo's flake](https://github.com/iffse/pay-respects/blob/main/flake.nix)) in my `home.nix` and i got a rebuild error.

<details>

<summary>`nix log /nix/store/ys249ajvmjwgrvsdlx6iw2n0fbjgchna-pay-respects-0.6.6.drv`</summary>

```log
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/azdvcx3fkh98bjq8a6ii9cz5bhzc3l89-source
source root is source
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 16 --target x86_64-unknown-linux-gnu --offline --profile release
[1m[31merror[0m[1m:[0m failed to load manifest for workspace member `/build/source/core`
referenced by workspace at `/build/source/Cargo.toml`

Caused by:
  failed to load manifest for dependency `pay-respects-parser`

Caused by:
  failed to read `/build/source/parser/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```

</details>

**Comment by ALameLlama** [comment]
@minhtrancccp I'm not entirely sure about building pay-respects via a nix flake, I installed it via the package manager https://search.nixos.org/packages?channel=24.11&show=pay-respects&from=0&size=50&sort=relevance&type=packages&query=pay-respects

This PR is just to add options to run `eval "$(pay-respects zsh --alias)"` so you can access it via `f`

**Comment by SigmaSquadron** [comment]
> I'm not sure if @SigmaSquadron has any interest in maintaining this also as they are the ones that added into nixpkgs

I'd maintain it if https://github.com/NixOS/rfcs/pull/182 is accepted and `home-manager` is merged/re-created in Nixpkgs, but not before.

**Comment by SigmaSquadron** [comment]
> i specified `pay-respects.package = inputs.pay-respect.packages.[system].default` (i.e. [the repo's flake](https://github.com/iffse/pay-respects/blob/main/flake.nix)) in my `home.nix` and i got a rebuild error.
> `nix log /nix/store/ys249ajvmjwgrvsdlx6iw2n0fbjgchna-pay-respects-0.6.6.drv`

Upstream issue. Looks like their build stopped working at some point. (Nixpkgs updates do take time, but they're usually reliable and well tested.)

File: modules/lib/maintainers.nix [hunk 1]
```diff
@@ -5,7 +5,6 @@
 # are expected to be follow the same format as described in [1].
 #
 # [1] https://github.com/NixOS/nixpkgs/blob/fca0d6e093c82b31103dc0dacc48da2a9b06e24b/maintainers/maintainer-list.nix#LC1
-
```
Comment (SigmaSquadron) [inline modules/lib/maintainers.nix:8]: This line was accidentally deleted.

File: modules/programs/pay-respects.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.pay-respects;
+  payRespectsCmd = "${cfg.package}/bin/pay-respects";
```
Comment (SigmaSquadron) [inline modules/programs/pay-respects.nix:5]:
```suggestion
  payRespectsCmd = getExe cfg.package;
```

File: modules/programs/pay-respects.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+with lib;
```
Comment (SigmaSquadron) [inline modules/programs/pay-respects.nix:2]:
```suggestion
```
Avoid nesting `with` expressions.

File: modules/programs/pay-respects.nix [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  options.programs.pay-respects = {
+    enable = mkEnableOption "pay-respects";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pay-respects;
+      defaultText = literalExpression "pkgs.pay-respects";
+      description = "The package to use for the pay-respects binary.";
+    };
```
Comment (SigmaSquadron) [inline modules/programs/pay-respects.nix:17]:
```suggestion
    package = mkPackageOption pkgs "pay-respects" { };
```

File: modules/programs/pay-respects.nix (hunk 3 above)
Comment (ALameLlama) [reply modules/programs/pay-respects.nix:2]: Thanks for taking the time to do that code review. I've fixed most of these and haven't pushed them yet.

Could you tell me if you're referring to doing something like this?

```
let
  inherit (lib) mkEnableOption mkPackageOption getExe optionalString mkIf;

  cfg = config.programs.pay-respects;
  payRespectsCmd = getExe cfg.package;
in {
```

Similar to https://github.com/nix-community/home-manager/blob/master/modules/programs/poetry.nix

Comment (ALameLlama) [reply modules/programs/pay-respects.nix]: I believe using inherit was correct and I've updated it

**Comment by ALameLlama** [comment]
Thanks for taking the time to do a code review, I've made those changes :D

File: modules/programs/pay-respects.nix (hunk 3 above)
Comment (SigmaSquadron) [reply modules/programs/pay-respects.nix]: Yeah, `inherit` is ideal IMO. https://nix.dev/guides/best-practices.html#with-scopes

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**